consumers/consumer.go

```diff
@@ -3,9 +3,18 @@
 
 package consumers
 
-// Consumer specifies message consuming API.
-type Consumer interface {
-	// Consume method is used to consumed received messages.
+// AsyncConsumer specifies asynchronous message consuming API.
+type AsyncConsumer interface {
+	// Consume method is used to consume received messages.
+	ConsumeAsync(messages interface{})
+
+	// An error channel is used to handle errors, so it supports the async approach.
```

Comment:
It's worth mentioning that the Error channel may receive `nil` error to indicate the success of async execution.